vizro-core/src/vizro/actions/_filter_action.py

```diff
@@ -38,12 +38,6 @@ def function(self, _controls: _Controls) -> dict[ModelID, Any]:
 
     @property
     def outputs(self):
-        # This is identical to _on_page_load.
-        outputs = {}
-
-        for target in self.targets:
-            component_id = target
-            component_property = cast(FigureType, model_manager[target])._output_component_property
-            outputs[target] = f"{component_id}.{component_property}"
-
-        return outputs
+        return {
+            target: cast(FigureType, model_manager[target])._action_outputs["__default__"] for target in self.targets
+        }
```

Comment:
I didn't fully understand this:

```
When this change is made the outputs property for filter, parameter and on_page_load should just
become `return self.targets` or similar.
```

Is this what you were expecting? Or should this really just work with `self.targets`? Is this always supposed to return a dict by the way?

Reply:
> Is this always supposed to return a dict by the way?

We always work with dict-like outputs for builtin actions because then we don't have to care about the order of outputs we return when the action is triggered. 

> Is this what you were expecting?

As we now attach the `__default__` action output if the ModelID is propagated as the dictionary value, then the following should also work: 

```suggestion
        return {target: target for target in self.targets}
```

Let's apply this change to all builtin actions where it suits:
- filter_action
- parameter_action
- on_page_load
- filter_interaction

Reply:
Ah I see - Yes, that works! Thanks!!